benchmark/main.cpp

```diff
@@ -5,6 +5,52 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 // clang-format on
+#include "utils.h"
+
 #include <benchmark/benchmark.h>
+#include <cuda_runtime.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#endif
+
+namespace {
+
+std::string getHostName() {
+#if defined(__linux__)
+  constexpr int len = 128;
+  char name[len];
+  gethostname(name, len);
+  return std::string(name);
+#else
+  return "UNKNOWN";
+#endif
+}
+
+std::string getDeviceName() {
+  int dev_idx;
+  NVFUSER_CUDA_RT_SAFE_CALL(cudaGetDevice(&dev_idx));
+
+  cudaDeviceProp prop;
+  NVFUSER_CUDA_RT_SAFE_CALL(cudaGetDeviceProperties(&prop, dev_idx));
+
+  return std::string(prop.name);
+}
+
+} // namespace
+
+// Copied from BENCHMARK_MAIN with extra custom settings
```

Comment:
Expanded from the default macro (`BENCHMARK_MAIN`) to include the custom context properties (i.e., "Host" and "GPU")